lldb/include/lldb/Target/ThreadPlanStack.h

```diff
@@ -194,7 +230,23 @@ class ThreadPlanStackMap {
 
 private:
   Process &m_process;
-  using PlansList = std::unordered_map<lldb::tid_t, ThreadPlanStack>;
+  // We don't want to make copies of these ThreadPlanStacks, there needs to be
+  // just one of these tracking each piece of work.  But we need to move the
+  // work from "attached to a TID" state to "detached" state, which is most
+  // conveniently done by having separate containers for the two states.
+  // To do that w/o having to spend effort fighting with C++'s desire to invoke
+  // copy constructors, we keep a store for the real object, and then keep
+  // pointers to the objects in the two containers.  The store actually holds
+  // shared_ptrs to the objects, so that the pointers in the containers will stay
+  // valid.
+  
+  // Use a shared pointer rather than unique_ptr here to prevent C++ from trying
```

Comment:
This doesn't really make sense, regardless of whether and object is stored in a shared or unique pointer, if you're moving the smart pointerrr around, it will copy/move the pointer, not the underlying object. Do you mean here that ThreadPlanStack itself should be copyable?